src/frontend/view/view.ts

```diff
@@ -121,52 +127,59 @@ const viewMainLayout = () => {
   return result.join('')
 }
 
+const viewDataBaseError = () => {
+  const html = []
+  let errorText = ''
+  switch (state.systemErrorText) {
+    case 'TIMEOUT':
+      errorText =
+        'The databse took too long to respond. This is usually due to too many query results. Please try a more precise query.'
+      break
+    default:
+      errorText = 'A problem has occurred with the database.'
+  }
+  html.push(`
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <h1 class="govuk-heading-xl">${errorTitle}</h1>
+        <p class="govuk-body">${errorText}</p>
+        <p class="govuk-body">Please <a class="govuk-link" href="mailto:data-products-research@digital.cabinet-office.gov.uk">contact the Data Products team</a> if the problem persists.</p>
+      </div>
+    </div>`)
+  return html.join('')
+}
+
 const viewErrorBanner = () => {
   const html = []
   if (state.systemErrorText || state.userErrors.length > 0) {
     html.push(`
         <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="govuk-error-summary">`)
-    if (state.systemErrorText) {
-      let errorText = ''
-      switch (state.systemErrorText) {
-        case 'TIMEOUT':
-          errorText =
-            'The databse took too long to respond. This is usually due to too many query results. Please try a more precise query.'
-          break
-        default:
-          errorText = 'A problem has occurred with the database.'
-      }
+
+    if (state.userErrors.length > 0) {
       html.push(`
-          <h1 class="govuk-error-summary__title" id="error-summary-title">There is a problem</h1>
-          <p class="govuk-body">${errorText}</p>
-          <p>Please <a class="govuk-link" href="mailto:data-products-research@digital.cabinet-office.gov.uk">contact the Data Products team</a> if the problem persists.</p>
-      `)
-    } else {
-      if (state.userErrors.length > 0) {
-        html.push(`
             <h1 class="govuk-error-summary__title" id="error-summary-title">
               There is a problem
             </h1>
             <ul class="govuk-error-summary__list">
           `)
-        state.userErrors.forEach((userError) => {
-          switch (userError) {
-            case 'missingWhereToSearch':
-              html.push(`
+      state.userErrors.forEach((userError) => {
+        switch (userError) {
+          case 'missingWhereToSearch':
```

Comment:
Could we put these constants in an enum?

Reply:
added enums